docs/debug/warmboot_reset.md

```diff
@@ -0,0 +1,163 @@
+
+It looks like moving SOL after call to `AmdInitEarly()` would fix the issue,
+but then it gets printed after a relatively long period - user might think
+that platform isn't booting, also it almost immediately disappears.
+
+Debug version of AGESA
+----------------------
+
+Output of boot process with Debug version of AGESA (unrelated lines omitted
+  for clarity):
+```
+
+coreboot-4.8-1064-g7321246a43-dirty Wed Jul 25 12:28:25 UTC 2018 romstage starting...
+14-25-48Mhz Clock settings
+(...)
+PC Engines apu5
+coreboot build 20182507
+BIOS version 4.8-1064-g7321246a43-dirty
+(...)
+agesawrapper_amdinitreset() entry
+CBFS: 'Master Header Locator' located CBFS at [200:7fffc0)
+CBFS: Locating 'AGESA'
+CBFS: Found @ offset 5ffdc0 size e6e52
+CBFS: 'Master Header Locator' located CBFS at [200:7fffc0)
+CBFS: Locating 'AGESA'
+CBFS: Found @ offset 5ffdc0 size e6e52
+
+AmdInitReset: Start
+
+
+*** !!!AGESAcb_Agesa ***
+
+
+[TP:C0]
+    FCH Reset Data Block Allocation: [0x0], Ptr = 0x004001F0
+Fch OEM config in INIT RESET Done
+
+[TP:B1]
+
+
+coreboot-4.8-1064-g7321246a43-dirty Wed Jul 25 12:28:25 UTC 2018 romstage starting...
+14-25-48Mhz Clock settings
+(...)
+PC Engines apu5
+coreboot build 20182507
+BIOS version 4.8-1064-g7321246a43-dirty
+(...)
+agesawrapper_amdinitreset() entry
+CBFS: 'Master Header Locator' located CBFS at [200:7fffc0)
+CBFS: Locating 'AGESA'
+CBFS: Found @ offset 5ffdc0 size e6e52
+CBFS: 'Master Header Locator' located CBFS at [200:7fffc0)
+CBFS: Locating 'AGESA'
+CBFS: Found @ offset 5ffdc0 size e6e52
+
+AmdInitReset: Start
+
+
+*** !!!AGESAcb_Agesa ***
+
+
+[TP:C0]
+    FCH Reset Data Block Allocation: [0x0], Ptr = 0x004001F0
+Fch OEM config in INIT RESET Done
+
+[TP:B1]
+
+AmdInitReset: Start
+
+
+*** !!!AGESAcb_Agesa ***
+
+
+[TP:C0]
+
+AmdInitReset: End
+
+
+[TP:C1]
+
+[TP:C4]
+(...)
+```
+
+It is clear that platrofm resets before `AmdInitEarly()`, somewhere in `AmdInitReset()`.
+
+PMxC0 S5/Reset Status
+---------------------
+
+As the issue depends on state from which platform is booted I looked through power
+management registers in [BKDG for AMD Family 16h Models 30h-3Fh Processors, 52740 Rev 3.06](https://support.amd.com/techdocs/52740_16h_models_30h-3fh_bkdg.pdf).
+Registers of interest were those with `Cold reset` value, as they were remembered
+through resets and transitions to/from S5, but not after full power cycle. The most
+important one is PMxC0 S5/Reset Status:
+> This register shows the source of previous reset.
+This register is also defined in AGESA headers in coreboot repository:
+`#define FCH_PMIOxC0_S5ResetStatus          0xFED803C0ul`
+
+Reading content of PMxC0 right before printing sign of life after different ways
+of (re)booting the platform resulted in the following values:
+
+| Entering/leaving S5 | PMxC0 during SOL | Bits set                     |
+|---------------------|------------------|------------------------------|
+| rte_ctrl -rel *     | 0x00000800       | SlpS3ToLtdPwrGdEn            |
```

Comment:
@krystian-hebel rte_ctrl is our internal stuff. W please use coldboot or equivalent.

Reply:
done